### PR TITLE
Nffs example

### PR DESCRIPTION
**Pull request by koffes**
I have created a NFFS file system example for NRF52 devices. This example creates a file system in flash if it does not already exist and writes a file to it. For each target power cycle the file will be read and a counter in the file incremented. Hopefully, this example will make using NFFS much easier.

### CODE REVIEW

**Comment by rbtchc** [comment]
I believe that you'll have to fix your coding style first before this can be merged.
ref: http://docs.zephyrproject.org/contribute/contribute_guidelines.html#coding-style

**Comment by carlescufi** [comment]
@koffes First things first, you need checkpatch to pass. Try running it locally or check the errors generated by Shippable.

**Comment by carlescufi** [comment]
@koffes Also, the commit message must be preceded by a prefix. For example:

samples: fs: Add an NFFS sample

File: samples/ext/nffs/README.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Overview
+*********
+Set up a Newtron Flash File system (NFFS) on a Nordic Semiconductor NRF5x device.
+
+
+Requirements
+************
+* NRF5x development kit. (Has only been tested on boards: BOARD=nrf52_pca10040 and BOARD=nrf52840_pca10056).
+* A serial terminal for observing example output. 115200 baud, no parity, 8 data bits, 1 stop bit, no HW flow control.
+
```
Comment (dbkinder) [inline samples/ext/nffs/README.rst]: Add information about installing the nRF5x Command Line Tools (in particular, ``nrfjprog``) as a requirement. It looks like they're included when installing the SEGGER software, but you should clarify that here.  I found [this Nordic material](http://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.tools%2Fdita%2Ftools%2Fnrf5x_command_line_tools%2Fnrf5x_installation.html) that might be worth linking to as well.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by koffes** [comment]
Thank you for the feedback. I am held up by a few other tasks now, but will address these issues as soon as possible.

**Comment by nvlsianpu** [comment]
@koffes this PR need to use `cmake` instead of `make` now.

File: samples/ext/nffs/prj.conf [hunk 2]
```diff
@@ -0,0 +1,25 @@
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_SOC_FLASH_NRF5=y
+CONFIG_SOC_FLASH_NRF5_DEV_NAME="NRF5_FLASH"
```
Comment (nvlsianpu) [inline samples/ext/nffs/prj.conf]: remove `CONFIG_SOC_FLASH_NRF5_DEV_NAME` - it was replaced by device tree property - so now you should use macro auto-generated FLASH_DEV_NAME for calls to nRF5 flash driver.

File: samples/ext/nffs/prj.conf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+CONFIG_DEBUG=y
+CONFIG_PRINTK=y
+
+CONFIG_FILE_SYSTEM=y
+CONFIG_FILE_SYSTEM_NFFS=y
+
+CONFIG_FS_NFFS_FLASH_DEV_NAME="NRF5_FLASH"
```
Comment (nvlsianpu) [inline samples/ext/nffs/prj.conf]: =NRF5_FLASH_DRV_NAME instead

File: samples/ext/nffs/src/main.c [hunk 4]
```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2016 Nordic Semiconductor ASA
```
Comment (nvlsianpu) [inline samples/ext/nffs/src/main.c]: I belief this file is whole new - you can add your or your company as copyright owner.

File: samples/ext/nffs/Makefile [hunk 5]
```diff
@@ -0,0 +1,2 @@
+CONF_FILE = prj.conf
```
Comment (nvlsianpu) [inline samples/ext/nffs/Makefile]: zephyr migrated form make to cmake.

Cmake should looks like this:
```
# Standard Zephyr application boilerplate.
include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
project(NONE)

target_sources(app PRIVATE
    src/main.c
)

zephyr_link_libraries(
    NFFS
)
```

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

**Comment by nvlsianpu** [comment]
Please rebase you work on top of the master - looks like you start from quite old master.

**Comment by koffes** [comment]
Thank you for your comments. Yes, this work was done some time ago and consequently, a lot of alterations has to be done. Unfortunately, my efforts has been directed elsewhere. Perhaps someone else can continue on this PR?

**Comment by nvlsianpu** [comment]
@koffes - unfortunately we all have lot of works, so I don't expect that someone else will take over the PR. But thank for communicate this early.

**Comment by nashif** [comment]
outdated, closing. Reopen with updates/rebase if needed.

**Pull request closed without merging**